### PR TITLE
PROD-10260 - Fix: Cannot find Group meta boxes (WP Fusion)

### PR DESCRIPTION
**Pull request by sajib-bb**
### Jira Issue:
https://buddyboss.atlassian.net/browse/PROD-10260

### General Note
Keep all conversations related to this PR in the associated Jira issue(s). Do NOT add comment on this PR or edit this PR’s description.

### Notes to Developer
* Ensure the IDs (i.e. PROD-1) of all associated Jira issues are reference in this PR’s title
* Ensure that you have achieved the Definition of Done before submitting for review
* When this PR is ready for review, move the associate Jira issue(s) to “Needs Review” (or “Code Review” for Dev Tasks)

### Notes to Reviewer
* Ensure that the Definition of Done have been achieved before approving a PR
* When this PR is approved, move the associated Jira issue(s) to “Needs QA” (or “Approved” for Dev Tasks)

### CODE REVIEW

File: src/bp-core/admin/settings/compat/wp-fusion.php [hunk 1]
```diff
@@ -329,3 +369,290 @@ function bb_legacy_wpf_create_tag() {
...
+			'sanitize_callback' => 'bb_legacy_wpf_sanitize_group_tag_ids',
+			'get_value'         => function ( $group ) {
+				$group_id = is_object( $group ) && isset( $group->id ) ? (int) $group->id : 0;
+				return bb_legacy_wpf_group_settings( $group_id )['apply_tags'];
+			},
+			'get_extra_data'    => function ( $group ) {
+				$group_id = is_object( $group ) && isset( $group->id ) ? (int) $group->id : 0;
+				return bb_legacy_wpf_group_tag_extra_data( bb_legacy_wpf_group_settings( $group_id )['apply_tags'] );
+			},
+			'save_value'        => function ( $group, $value ) {
```
Comment (reza-bb) [inline src/bp-core/admin/settings/compat/wp-fusion.php:590]: Removing the last chip makes AjaxMultiSelectField emit [], appendToFormData() in utils/ajax.js appends nothing for an empty array, so registered_field_wpf_group_* is absent from $_POST and save_fields_data() skips this field entirely (if ( ! isset( $_POST[ $post_key ] ) ) { continue; }).
Net effect: these three fields are add/replace-only — "Link with Tag" can never be unset, and the value reappears on reopen. Removing one of two tags works; going to zero does not.
Smallest fix is in GroupEditModal.js:333 — send '' when the value is an empty array. bb_legacy_wpf_sanitize_group_tag_ids( '' ) already returns array(), so no PHP change needed.

Same for line #621 & #651

File: src/bp-core/admin/settings/compat/wp-fusion.php [hunk 2]
```diff
@@ -329,3 +369,290 @@ function bb_legacy_wpf_create_tag() {
...
+
+	// Match the legacy handler's own empty-settings cleanup
+	// (`WPF_BuddyPress::save_groups_data()`): once every leaf is empty, delete
+	// the row entirely instead of leaving `a:3:{...}` of empty arrays behind.
+	if ( empty( array_filter( $settings ) ) ) {
+		groups_delete_groupmeta( $group_id, 'wpf-settings-buddypress' );
+		return;
+	}
+
+	groups_update_groupmeta( $group_id, 'wpf-settings-buddypress', $settings );
```
Comment (reza-bb) [inline src/bp-core/admin/settings/compat/wp-fusion.php:462]: Direct meta write bypasses WP Fusion's own group save

`compat/bp-auto-group-join.php` replays the legacy side effect on `bb_admin_after_save_group` precisely because the plugin's own nonce path never runs in Settings 2.0. This writes groupmeta directly and never populates `$_POST['wpf-settings-buddypress']`, while `class-bb-admin-groups-ajax.php:1475` still fires `bp_group_admin_edit_after`.

Two risks: any member/tag reconciliation inside `WPF_BuddyPress::save_groups_data()` no longer fires, and if that handler writes/deletes the meta key unconditionally, it could clobber what we just saved.
Can't verify WP Fusion's source from this repo—please confirm the hook and body, and re-populate the array into $_POST (or call the sync explicitly) if there are side effects.

**Additional Recommendation:** Follow the documented precedent in the sibling module: `bp-auto-group-join.php` registers on `bb_admin_after_save_group` at a late priority and replays the legacy side effect from the final saved meta. Confirm WP Fusion's actual hook and handler body, then either replay `wpf-settings-buddypress` into `$_POST` before that action or call the sync explicitly. If the investigation shows `save_groups_data()` is purely a meta writer with no side effects, that's a fine outcome; but it needs to be stated in the DocBlock, not left silent.

File: src/bp-core/admin/settings/compat/wp-fusion.php [hunk 3]
```diff
@@ -329,3 +369,290 @@ function bb_legacy_wpf_create_tag() {
...
+ *
+ * @param BB_Admin_Meta_Field_Registry $registry  Registry instance.
+ * @param string                       $component Component identifier.
+ */
+function bb_legacy_wpf_register_group_fields( $registry, $component ) {
+	if ( 'groups' !== $component ) {
+		return;
+	}
+
+	if ( ! function_exists( 'groups_get_groupmeta' ) ) {
```
Comment (reza-bb) [inline src/bp-core/admin/settings/compat/wp-fusion.php:558]: No guard that WP Fusion's BuddyPress integration is loaded

File: src/bp-core/admin/settings/compat/wp-fusion.php [hunk 4]
```diff
@@ -329,3 +369,290 @@ function bb_legacy_wpf_create_tag() {
...
+		$value = trim( (string) sanitize_text_field( $value ) );
+		if ( '' !== $value ) {
+			$out[] = $value;
+		}
+	}
+
+	$out = array_values( array_unique( $out ) );
+
+	if ( null !== $max && $max > 0 && count( $out ) > $max ) {
+		$out = array_slice( $out, -$max );
```
Comment (reza-bb) [inline src/bp-core/admin/settings/compat/wp-fusion.php:492]: silent truncation on the single-tag fields

File: src/bp-core/admin/settings/compat/wp-fusion.php [hunk 5]
```diff
@@ -329,3 +369,290 @@ function bb_legacy_wpf_create_tag() {
...
+	if ( ! is_array( $raw ) ) {
+		$raw = ( '' === $raw || null === $raw ) ? array() : array( $raw );
+	}
+
+	$out = array();
+	foreach ( $raw as $value ) {
+		if ( ! is_scalar( $value ) ) {
+			continue;
+		}
+		$value = trim( (string) sanitize_text_field( $value ) );
```
Comment (reza-bb) [inline src/bp-core/admin/settings/compat/wp-fusion.php:483]: sanitize_text_field() can mutate add_tags-CRM tag ids

File: src/bp-core/admin/settings/compat/wp-fusion.php [hunk 6]
```diff
@@ -329,3 +369,290 @@ function bb_legacy_wpf_create_tag() {
...
+	);
+
+	$group_id = (int) $group_id;
+	if ( $group_id <= 0 ) {
+		return $defaults;
+	}
+
+	$stored = groups_get_groupmeta( $group_id, 'wpf-settings-buddypress' );
+
+	return is_array( $stored ) ? array_merge( $defaults, $stored ) : $defaults;
```
Comment (reza-bb) [inline src/bp-core/admin/settings/compat/wp-fusion.php:420]: array_merge( $defaults, $stored ) guarantees the three keys exist but preserves whatever type is stored. If a leaf were ever a scalar, get_value returns a string, Array.isArray() fails and the saved tag renders as no chip.
It does self-heal on the next save, so this is hardening rather than a bug.
A (array) cast on each leaf in the return would close it.

**Review by reza-bb — COMMENTED** [review]
Please check the feedback and update accordingly.

File: src/js/admin/settings/components/groups/GroupEditModal.js [hunk 7]
```diff
@@ -330,7 +330,17 @@ export function GroupEditModal( { isOpen, group, onClose, onSave, isSaving } ) {
...
+				// (appendToFormData() has nothing to iterate), so the POST key
+				// is simply absent and save_fields_data() skips the field — an
+				// array-type field could never be cleared back to empty once
+				// it held a value. Send '' instead so the key reaches $_POST;
+				// each array field's sanitize_callback is expected to treat ''
+				// as "empty selection" (see e.g.
+				// bb_legacy_wpf_sanitize_group_tag_ids()).
+				var isEmptyArray = Array.isArray( val ) && 0 === val.length;
+
+				payload[ 'registered_field_' + field.id ] = ( null !== val && undefined !== val && ! isEmptyArray ) ? val : '';
```
Comment (reza-bb) [inline src/js/admin/settings/components/groups/GroupEditModal.js:343]: There are three copies of this payload-building logic. The empty-array fix was applied to one modal, not the shared payload builder. Check for similar occurrences and apply the same fix there too.

File: src/bp-core/admin/settings/compat/wp-fusion.php [hunk 8]
```diff
@@ -329,3 +369,330 @@ function bb_legacy_wpf_create_tag() {
...
+ * `bp-auto-group-join`'s 6000-6030 block, so if both plugins are active their
+ * sections appear in a stable, predictable sequence.
+ *
+ * @since BuddyBoss 3.1.0
+ *
+ * @param BB_Admin_Meta_Field_Registry $registry  Registry instance.
+ * @param string                       $component Component identifier.
+ */
+function bb_legacy_wpf_register_group_fields( $registry, $component ) {
+	if ( 'groups' !== $component ) {
```
Comment (reza-bb) [inline src/bp-core/admin/settings/compat/wp-fusion.php:586]: Instead of returning from this function in 3 different steps, use all the conditions in one block and return if they are not satisfied.

File: src/bp-core/admin/settings/compat/wp-fusion.php [hunk 9]
```diff
@@ -329,3 +369,330 @@ function bb_legacy_wpf_create_tag() {
...
+	$group_id = (int) $group_id;
+	if ( $group_id <= 0 ) {
+		return $defaults;
+	}
+
+	$stored   = groups_get_groupmeta( $group_id, 'wpf-settings-buddypress' );
+	$settings = is_array( $stored ) ? array_merge( $defaults, $stored ) : $defaults;
+
+	foreach ( $settings as $key => $value ) {
+		$settings[ $key ] = array_values( (array) $value );
```
Comment (reza-bb) [inline src/bp-core/admin/settings/compat/wp-fusion.php:426]: The (array) cast is applied to every key in the merged array, not only the three this bridge owns. 

`array_merge( $defaults, $stored )` deliberately preserves any extra keys WP Fusion stores under wpf-settings-buddypress, and the loop then coerces each of them with `array_values( (array) $value )`. Because `bb_legacy_wpf_save_group_setting()` writes the whole settings array back, a scalar sibling key; a future checkbox stored as true or a string- would be silently rewritten as array( true ) in the database on the first group save.

**Recommendation:** Restrict the coercion to the three known keys; iterate `array_keys( $defaults )` instead of `$settings` — so unrecognised keys pass through byte-identical on the read-modify-write round trip.

File: src/bp-core/admin/settings/compat/wp-fusion.php [hunk 10]
```diff
@@ -329,3 +369,330 @@ function bb_legacy_wpf_create_tag() {
...
+
+	// `groups_get_groupmeta()` only confirms BuddyPress groups are active, not
+	// that WP Fusion's own BuddyPress module is loaded — `WPF_BuddyPress`
+	// (includes/integrations/class-buddypress.php) is the class that owns
+	// `wpf-settings-buddypress` and is what these fields bridge against.
+	if ( ! class_exists( 'WPF_BuddyPress' ) ) {
+		return;
+	}
+
+	$group_label = 'WP Fusion - Group Settings';
```
Comment (reza-bb) [inline src/bp-core/admin/settings/compat/wp-fusion.php:602]: The section heading is not translatable

**Review by reza-bb — COMMENTED** [review]
Please check the feedback and work on it accordingly

File: src/bp-core/admin/settings/compat/wp-fusion.php (hunk 2 above)
Comment (reza-bb) [reply src/bp-core/admin/settings/compat/wp-fusion.php:462]: Please work on this feedback

File: src/js/admin/settings/components/groups/GroupEditModal.js (hunk 7 above)
Comment (sajib-bb) [reply src/js/admin/settings/components/groups/GroupEditModal.js:343]: fix applied

File: src/bp-core/admin/settings/compat/wp-fusion.php (hunk 8 above)
Comment (sajib-bb) [reply src/bp-core/admin/settings/compat/wp-fusion.php]: fix applied

File: src/bp-core/admin/settings/compat/wp-fusion.php (hunk 9 above)
Comment (sajib-bb) [reply src/bp-core/admin/settings/compat/wp-fusion.php]: fix applied

File: src/bp-core/admin/settings/compat/wp-fusion.php (hunk 10 above)
Comment (sajib-bb) [reply src/bp-core/admin/settings/compat/wp-fusion.php]: fix applied

File: src/bp-core/admin/settings/compat/wp-fusion.php (hunk 2 above)
Comment (reza-bb) [reply src/bp-core/admin/settings/compat/wp-fusion.php]: This is the one item from the **3 Sept review** that still has no commit or reply. Writing the meta directly means `WPF_BuddyPress::save_groups_data()` never runs (no $_POST['wpf-settings-buddypress'], and the `bp_group_admin_edit_after` call is wrapped in try/catch so a nonce failure is silent). If that handler reconciles member tags when "Link with Tag" changes, that behaviour is now lost; if it writes unconditionally, it can overwrite what we just saved. 

Please read the handler in WP Fusion and then either replay the saved array on `bb_admin_after_save_group` the way `bp-auto-group-join.php` does, or state in this docblock (with the WP Fusion version you checked) that it's a pure meta writer with no side effects.  If you go with the second option, the test the corner cases properly. I can't approve without one of those two.

**Review by reza-bb — COMMENTED** [review]
Please check the feedback that you missed twice and work on it accordingly.
